docs/general/user-defined-functions.rst

```diff
@@ -197,31 +197,48 @@ Optionally, you can provide a schema::
 Supported languages
 ===================
 
-CrateDB currently only supports the UDF language ``javascript``.
+CrateDB currently only supports the ``JavaScript`` user-defined functions.
 
 .. _udf_lang_js:
 
 JavaScript
 ----------
 
-The UDF language ``javascript`` supports the `ECMAScript 5.1`_ standard.
+The user defined function JavaScript is compatible with the `ECMAScript 2019`_
+specification.
 
 .. NOTE::
 
    The JavaScript language is an :ref:`enterprise feature
    <enterprise-features>`.
 
-CrateDB uses the Java built-in JavaScript engine Nashorn_ to interpret and
-execute functions written in JavaScript. The engine is initialized using the
-``--no-java`` option which basically restricts all access to Java APIs from
-within the JavaScript context. CrateDB's engine also does not allow
-non-standard syntax extensions (``--no-syntax-extensions``).
+CrateDB uses the `GraalVM JavaScript`_ engine as a JavaScript (ECMAScript)
+language execution runtime. The `GraalVM JavaScript`_ engine is a Java
+application that works on the stock Java VMs. The interoperability between
+Java code (host language) and JavaScript user-defined functions
+(guest language) is guaranteed thtough the GraalVM `GraalVM Polyglot API`_.
+
+Please note that CrateDB does not use the GraalVM JIT compiler as the
+optimizing compile of the GraalVM JavaScript code as one might expect while
+seeing usage of the GraalVM. The `stock host Java VM JIT compilers`_ still can
+JIT-compile, execute and optimize the GraalVM JavaScript codebase to certain
+extend.
+
+The execution context for guest JavaScript language code is created with
+restricted privileges to allow for the execution of less trusted guest
+language code. The guest language application context for each user-defined
+function, that controls which resources of the host language can be accessed,
+is created with default access modifiers meaning that the access to all
+managed resources is denied. The exception is only the host language
+interoperability configuration that explicitly allows the access to Java lists
+and arrays. Please refer to `GraalVM Security Guide`_ for more detailed
+information.
```

Comment:
how about:

> The execution context for guest JavaScript is created with restricted privileges to allow for the safe execution of less trusted guest language code. The guest language application context for each user-defined function is created with default access modifiers, so any access to managed resources is denied. The only exception is the host language interoperability configuration which explicitly allows access to Java lists and arrays. Please refer to `GraalVM Security Guide`_ for more detailed information.

this reads a little clearer to me. but have I preserved the correct meaning with my edit?

Reply:
> The only exception is the host language interoperability configuration which explicitly allows access to Java lists and arrays. 

I was trying to say that all access configuration of the guest language context are set to default, except the host language interoperability configuration where we had to allow access to objects and lists of the host language.